pkg/common/proto/agent.proto

```diff
@@ -0,0 +1,50 @@
+syntax = "proto3";
+
+service AgentService{
+    rpc PerformFantasyActions(FantasyLandscape) returns (FantasySelections) {}
+}
+
+message FantasyLandscape{
+    LeagueSettings settings = 1;
+    FantasyTeam botTeam = 2;
+    repeated Player players = 3;
+    /* TODO historical / statistical data */
+}
+
+message FantasySelections{
+    repeated PlayerSlot slots = 1;
+}
+
+message FantasyTeam{
+    int32 id = 1;
+    string name = 2;
+    string owner = 3;
+}
+
+message LeagueSettings{
+    int32 numTeams = 1;
+    repeated PlayerSlot slotsPerTeam = 2;
+    /* scoring, budgets, etc. */
+}
+
+message PlayerSlot{
+    string name = 1;
+    bool canBeEmpty = 2;
+    repeated string allowedPlayerPositions = 4;
+    optional int32 assignedPlayerId = 3;
+}
+
+enum PlayerStatus {
+    PLAYER_STATUS_ACTIVE = 0;
+    PLAYER_STATUS_QUESTIONABLE = 1;
+    PLAYER_STATUS_OUT = 2;
+}
+
+message Player{
+    int32 id = 1;
+    string name = 2;
+    repeated string allowedPositions = 3;
+    string team = 4; /* professional team, not fantasy team */
+    PlayerStatus status = 5;
```

Comment:
Btw, here is the example response from nflreadr::load_injuries()

```
<div class="dataTables_scrollHead" style="background-color: rgb(255, 255, 255); background-image: none !important; background-position: initial !important; background-size: initial !important; background-repeat: initial !important; background-attachment: initial !important; background-origin: initial !important; background-clip: initial !important; color: rgb(0, 0, 0); font-family: &quot;DejaVu Sans&quot;, &quot;Lucida Grande&quot;, &quot;Segoe UI&quot;, Verdana, Helvetica, sans-serif; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; overflow: hidden; position: relative; border: 0px; width: 844px;"><div class="dataTables_scrollHeadInner" style="box-sizing: content-box; width: 1901px; padding-right: 0px;">
season | game_type | team | week | gsis_id | position | full_name | first_name | last_name | report_primary_injury | report_secondary_injury | report_status | practice_primary_injury | practice_secondary_injury | practice_status | date_modified
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --


</div>
```

Reply:
Should we leave player status as a string then and allow this to be specified based on the data set?

I'm not sure it makes sense to include multiple injury fields this early.

Reply:
Yeah string is good.